extension/js/common/platform/store/contact-store.ts

```diff
@@ -105,24 +106,23 @@ export class ContactStore extends AbstractStore {
       }
       // X.509 certificate
       if (PgpKey.getKeyType(pubkey) === 'x509') {
-        // FIXME: For now we return random data.
-        // Later we'll return serial ID from the certificate.
-        const longid = Math.random() + '';
+        const fingerprint = smime.getSerialNumber(pubkey);
+        const longid = fingerprint.substring(fingerprint.length / 2);
         return {
           email: validEmail,
           name: name || null,
           pubkey,
           has_pgp: 1, // number because we use it for sorting
           searchable: ContactStore.dbCreateSearchIndexList(validEmail, name || null, true),
           client: ContactStore.storablePgpClient(client || 'pgp'),
-          fingerprint: Math.random() + '',
+          fingerprint,
           longid,
           longids: [longid],
           pending_lookup: 0,
           last_use: lastUse || null,
           pubkey_last_sig: lastSig || null,
           pubkey_last_check: lastCheck || null,
-          expiresOn: null
+          expiresOn: smime.getExpirationTime(pubkey) || null
```

Comment:
While the Date will be a bit less convenient here (need to convert to number), I still think it's generally better to work with a Date.